modules/trusted-firmware-m/Kconfig.tfm

```diff
@@ -204,6 +204,19 @@ config TFM_IMAGE_VERSION_NS
 	  versions of both the secure firmware and non-secure firmware. This sets
 	  the non-secure firmware's version for rollback prevention.
 
+config TFM_IMAGE_SECURITY_COUNTER
+	int "Security counter value used for hardware rollback protection"
+	depends on TFM_BL2
+	range 1 1024
+	default 1
+	help
+	  By default, TFM enables hardware rollback protection, which requires a security counter
+	  to be embedded in the image trailer. The default behaviour of constructing this counter
+	  from the image version breaks the TFM `boot_nv_security_counter_update` implementation
+	  once the version number is greater than `0.0.1024`. As such, explicitly specify the
+	  desired security counter value. As per the MCUboot docs, this does not need to be
+	  incremented on every firmware update.
```

Comment:
Can you link to the actual documentation that says this?

Reply:
https://github.com/zephyrproject-rtos/mcuboot/blob/main/docs/design.md#hardware-based-downgrade-prevention
If you instead meant putting it in the kconfig, I'm not sure what the policy there is, as the link could in theory change at any time.

Reply:
Okay so it doesn't need to be incremented on every release, but it's by incrementing it that the rollback protection is effective. So maybe I wouldn't say that in the description.
And I don't know about such policies, but I'd say that you could put the link in there if you take a particular Git commit or tag (rather than `main`) because it's a Zephyr-controlled repository.

Reply:
I feel `https://github.com/zephyrproject-rtos/mcuboot/blob/52e2afc2f809c424b0f337f56059d1dfcc7e6d98/docs/design.md#hardware-based-downgrade-prevention` is going to make some linters very unhappy. It's about twice the column limit by itself.

Reply:
From my limited experience the linters let long URLs pass. Or then, another option would be to just put `"Hardware-based downgrade prevention" in mcuboot/docs/design.md`. Anyhow, I think it would be nice to one way or another point to the documentation referenced.